application/F3DStarter.cxx

```diff
@@ -1636,4 +1639,17 @@ void F3DStarter::AddCommands()
         this->LoadFileGroup(index);
       }
     });
+    interactor.addCommand("open_file_dialog", 
+      [this](const std::vector<std::string>&)
+      {
+         char* file = tinyfd_openFileDialog( "Open File", ".", 0, nullptr, nullptr, false);
+         if (file) 
+         {
+           int index = this->AddFile(file);
+           if (index > -1) 
+           {
+             this->LoadFileGroup(index);
+           }
+         }
```

Comment:
Here a suggestion. You need to define `CTEST_OPEN_DIALOG_FILE` (see other comment).
Wdyt @mwestphal ?

```suggestion
         std::string filePath;
         const char* testFile = getenv("CTEST_OPEN_DIALOG_FILE");
         if (testFile)
         {
           filePath = testFile;
         }
         else
         {
           char* file = tinyfd_openFileDialog( "Open File", ".", 0, nullptr, nullptr, false);
           if (file) 
           {
             filePath = file;
           }
         }

         if (!filePath.empty())
         {
           int index = this->AddFile(file);
           if (index > -1) 
           {
             this->LoadFileGroup(index);
           }
         }
```

Reply:
I implemented this, however this will not be able to test the functionality of tinyfiledialogs, it is only testing the functionality of AddFile and LoadFileGroup. It won't be able to catch if tinyfiledialogs ever breaks

Reply:
yeah, but I dont think this is actually testable. Its better then nothing though.